assets/wizards/popups/views/segments/SegmentsList.js

```diff
@@ -304,47 +336,93 @@ const SegmentsList = ( { wizardApiFetch, segments, setSegments, isLoading } ) =>
-						totalSegments={ segments.length }
-					/>
-				) ) }
-			</div>
-		</Fragment>
-	) : (
-		<Fragment>
-			<Card headerActions noBorder>
-				<h2>{ __( 'You have no saved audience segments.', 'newspack' ) }</h2>
-				<AddNewSegmentLink />
-			</Card>
-			<p>
-				{ __(
-					'Create audience segments to target visitors by engagement, activity, and more.',
-					'newspack'
+	return (
+		<>
+			{ segments.length ? (
+				<>
+					{ error && <Notice noticeText={ error } isError /> }
+					<Card headerActions noBorder>
+						<h2>{ __( 'Audience segments', 'newspack' ) }</h2>
+						<AddNewSegmentLink />
+					</Card>
+					<div
+						className={
+							'newspack-campaigns-wizard-segments__list' + ( inFlight ? ' is-loading' : '' )
+						}
+						ref={ ref }
+					>
+						{ segmentsToShow.map( ( segment, index ) => (
+							<SegmentActionCard
+								deleteSegment={ deleteSegment }
+								key={ segment.id }
+								inFlight={ inFlight || isLoading > 0 }
+								segment={ segment }
+								segments={ segments }
+								sortSegments={ sortSegments }
+								index={ index }
+								wrapperRef={ ref }
+								dropTargetIndex={ dropTargetIndex }
+								setDropTargetIndex={ setDropTargetIndex }
+								totalSegments={ segments.length }
+							/>
+						) ) }
+					</div>
+				</>
+			) : (
+				<>
+					<Card headerActions noBorder>
+						<h2>{ __( 'You have no saved audience segments.', 'newspack' ) }</h2>
+						<AddNewSegmentLink />
+					</Card>
+					<p>
+						{ __(
+							'Create audience segments to target visitors by engagement, activity, and more.',
+							'newspack'
+						) }
+					</p>
+				</>
+			) }
+			{ showDefaultModal && (
+				<Modal
+					title={ __( 'Default segments', 'newspack' ) }
+					onRequestClose={ () => setShowDefaultModal( false ) }
+				>
+					<p>
+						{ sprintf(
+							// Translators: help message before generating default segments.
+							__(
+								'This will %1$s set of default segments that represent a basic engagement funnel.%2$s Proceed?',
+								'newspack'
+							),
+							defaultSegments ? __( 'regenerate the', 'newspack' ) : __( 'generate a', 'newspack' ),
+							defaultSegments
+								? __(
+										' Any changes you’ve made to the default segments will be ovewritten.',
+										'newspack'
+								  )
+								: ''
+						) }
+					</p>
+					<Card buttonsCard noBorder className="justify-end">
+						<Button isSecondary onClick={ () => setShowDefaultModal( false ) }>
+							{ __( 'Cancel', 'newspack' ) }
+						</Button>
+						<Button isPrimary onClick={ () => generateDefaultSegments() }>
+							{ __( 'OK', 'newspack' ) }
+						</Button>
+					</Card>
+				</Modal>
+			) }
+			<Button isPrimary onClick={ () => setShowDefaultModal( true ) }>
+				{ sprintf(
+					// Translators: button label for generating or resetting the default segments.
+					__( '%s default segments', 'newspack' ),
+					defaultSegments || defaultsCreated
```

Comment:
Ah, this is because the global we're reading is empty and the state gets cleared after unmounting the view. I added a check via API if the global is empty to handle this case: 6515f25